erpnext/hr/doctype/additional_salary/additional_salary.py

```diff
@@ -21,10 +21,23 @@ def validate(self):
 			frappe.throw(_("Amount should not be less than zero."))
 
 	def validate_dates(self):
- 		date_of_joining, relieving_date = frappe.db.get_value("Employee", self.employee,
+		date_of_joining, relieving_date = frappe.db.get_value("Employee", self.employee,
 			["date_of_joining", "relieving_date"])
- 		if date_of_joining and getdate(self.payroll_date) < getdate(date_of_joining):
- 			frappe.throw(_("Payroll date can not be less than employee's joining date"))
+
+		if not self.is_recurring and not self.payroll_date:
+			frappe.msgprint(_("Please enter Payroll Date."), indicator='blue', raise_exception=1)
+		if self.is_recurring and not self.from_date and not self.to_date:
```

Comment:
Use "Mandatory Depends On" property for such validation.

"To Date" should be optional, many times user will not know the End Date, "Allow on Submit" should be enabled.

Reply:
done